components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -1662,16 +1663,26 @@ private static void buildSAMLAuthenticationConfiguration(IdentityProvider fedIdp
         String authenticationContextClass = paramMap.get(
                 IdentityApplicationConstants.Authenticator.SAML2SSO.AUTHENTICATION_CONTEXT_CLASS);
 
-        if (IdentityApplicationConstants.Authenticator.SAML2SSO.
-                CUSTOM_AUTHENTICATION_CONTEXT_CLASS_OPTION.equals(authenticationContextClass)) {
-            authenticationContextClass = paramMap.get(IdentityApplicationConstants.
-                    Authenticator.SAML2SSO.ATTRIBUTE_CUSTOM_AUTHENTICATION_CONTEXT_CLASS);
-        }
         property = new Property();
         property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.AUTHENTICATION_CONTEXT_CLASS);
         property.setValue(authenticationContextClass);
         properties.add(property);
 
+        String customAuthenticationContextClass = "";
+        String[] authnContextClassList = authenticationContextClass.split(",");
+
+        if (Arrays.asList(authnContextClassList).contains(
+            IdentityApplicationConstants.Authenticator.SAML2SSO.CUSTOM_AUTHENTICATION_CONTEXT_CLASS_OPTION)) {
+            customAuthenticationContextClass = paramMap.get(IdentityApplicationConstants.
+                    Authenticator.SAML2SSO.ATTRIBUTE_CUSTOM_AUTHENTICATION_CONTEXT_CLASS);
+        }
+
+        property = new Property();
+        property.setName(IdentityApplicationConstants.
+                Authenticator.SAML2SSO.ATTRIBUTE_CUSTOM_AUTHENTICATION_CONTEXT_CLASS);
+        property.setValue(customAuthenticationContextClass);
+        properties.add(property);
```

Comment:
Let's add this part as well as the customAuthenticationContextClass variable instantiation to the if block above as it only needed to execute if a custom auth context class is configured.

Reply:
default flow can handle seperately